crates/client/rpc-core/src/trace.rs

```diff
@@ -0,0 +1,183 @@
+// These are conversion utils for going from StarknetApi/Blockifier (used in runtime) to
+// StarknetCore (used for rpc)
+
+use blockifier::execution::entry_point::{CallInfo, CallType as BlockifierCallType, MessageToL1, Retdata};
+use mp_starknet::transaction::types::TransactionExecutionInfoWrapper;
+use starknet_api::api_core::{ClassHash, ContractAddress, EntryPointSelector};
+use starknet_api::deprecated_contract_class::EntryPointType;
+use starknet_api::hash::StarkFelt;
+use starknet_api::transaction::{Calldata, EthAddress, EventContent};
+use starknet_core::types::{Event, FeeEstimate, FieldElement, MsgToL1};
+
+use crate::types::{
+    CallType, DeclareTransactionTrace, DeployAccountTransactionTrace, ExecuteInvocation, ExecutionError,
+    FunctionInvocation, InvokeTransactionTrace, L1HandlerTransactionTrace, SimulateTransactionResult, TransactionTrace,
+};
+
+/// Aggregates fee estimation from execution info (similar to what estimate_fee does)
+pub fn get_fee_estimate(execution_info: &TransactionExecutionInfoWrapper) -> FeeEstimate {
+    FeeEstimate {
+        overall_fee: execution_info.actual_fee.0 as u64,
+        gas_price: 0,
+        // TODO: actual_resources are not passed atm because it's tricky to implement SCALE on hashmap
+        // Alternatively we can aggregate consumed gas from all the call_info structures
+        gas_consumed: 0,
+    }
+}
+
+/// Converts StarkFelt used in Blockifier/StarknetAPI to FieldElement from starknet-rs
+pub fn convert_felt(felt: StarkFelt) -> FieldElement {
+    FieldElement::from_byte_slice_be(felt.bytes()).expect("Expected just to be diff representations of [u8; 32]")
+}
```

Comment:
Don't we have an into for that ? if not plz implement the from trait

Reply:
All this will probably be change with my big type PR.
Wait for me to merge it and then do this job. But it won't be on the same types so, don't bother with if for now

Reply:
So what's the resolution, keeping these casts as is and remove later or implement `From` in starknet-api / blockifier (that would introduce additional dependency to those crates)?

Reply:
The best case is my PR is merged at the end of the week and you rebase on it next week.
If you don'y work on it full time and start with the other comments +  those on the other related PRs, it should be good. I hope